src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -7865,6 +7865,7 @@ def astype(
         self,
         col_dtypes_map: dict[str, Union[dtype, ExtensionDtype]],
         errors: Literal["raise", "ignore"] = "raise",
+        is_index: bool = False,
```

Comment:
maybe call it include_index to indicate wether to include the index for dtypes?

Reply:
I named it `is_index` because it is not doing index columns along with data columns, it's only doing index columns. Only the Index object uses this.